distribution/client/src/lib/update.ts

```diff
@@ -79,24 +87,39 @@ export default class Update {
     return this.app.service('update/tainted').create({
       uuid: this.uuid,
       level: toBeTaintedLevel
-    }, {});
+    }, {})
+    .catch( (e) => {
+      logger.error('Tainting went wrong!', e);
+      throw e;
+    });
   }
   /*
-   * Apply the updates provided by the given Update Manifest
-   *
+   * Apply the updates provided by the given Update Manifest.
+   * Rolls back automatically to the previous non tainted level if failing.
+   * (Note: the rollback may not end up on most of the time, that would mean going from say UL-0 to U)
+
+   1) UL10 gets globally tainted after an issue is identified (UL9 is not tainted)
+   2) Instance attempts updating from UL10 to UL11, and fails
+   3) update.query() to get the right updates => receives UL9...
+   4) config files changed from UL9 to UL10, Jenkins fails to start after rolling back.
+
    * @param  {Map} Update Manifest described in JEP-307
    * @return {Promise} Which resolves once updates have been applied
    * @return {boolean} False if there is already an update in progress
    */
-  async applyUpdates(updates) {
-    if (this.updateInProgress || (!updates)) {
+  async applyUpdates(updates, forceUpdate?: boolean) {
+    if (forceUpdate) {
+      logger.warn('Forced update (expected during a rollback)');
```

Comment:
During a force update, shouldn't the `updateInProcess` still be assigned a value?

Reply:
Good point. Normally, though it's indeed not enforced, the current code is expecting that calling force is used when calling itself recursively for rolling back, so `updateInProcess` should already have a value. 

Going to have a look, probably will mainly add a warning if not.